src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/set_global_properties_request.cc

```diff
@@ -820,9 +848,130 @@ bool SetGlobalPropertiesRequest::IsWhiteSpaceExist() {
+            .asInt());
+  }
+
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+  auto saved_keyboard_props = app->keyboard_props();
+  if (saved_keyboard_props) {
+    if (saved_keyboard_props->keyExists(hmi_request::keyboard_layout)) {
+      return static_cast<hmi_apis::Common_KeyboardLayout::eType>(
+          (*saved_keyboard_props)[hmi_request::keyboard_layout].asInt());
+    }
+  }
+
+  return hmi_apis::Common_KeyboardLayout::QWERTY;
+}
+
+uint32_t SetGlobalPropertiesRequest::GetAllowedNumberOfConfigurableKeys()
+    const {
+  SDL_LOG_AUTO_TRACE();
+
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+  auto display_capabilities = app->display_capabilities();
+  if (!display_capabilities) {
+    SDL_LOG_WARN("Display capabilities are not available");
+    return 0;
+  }
+
+  auto* window_capabilities =
+      (*display_capabilities)[0][strings::window_capabilities].asArray();
+
+  if (!window_capabilities) {
+    SDL_LOG_WARN("Window capabilities are not available");
+    return 0;
+  }
+
+  if (!(*window_capabilities)[0].keyExists(
+          hmi_response::keyboard_capabilities)) {
+    SDL_LOG_WARN("Keyboard capabilities are not available");
+    return 0;
+  }
+
+  if (!(*window_capabilities)[0][hmi_response::keyboard_capabilities].keyExists(
+          hmi_response::configurable_keys)) {
+    SDL_LOG_WARN("Configurable keys are not available");
+    return 0;
+  }
+
+  auto configurable_keyboards =
+      (*window_capabilities)[0][hmi_response::keyboard_capabilities]
+                            [hmi_response::configurable_keys]
+                                .asArray();
+
+  const auto requested_layout = GetKeyboardLayout();
+  for (auto keyboard : (*configurable_keyboards)) {
+    if (requested_layout ==
+        static_cast<hmi_apis::Common_KeyboardLayout::eType>(
+            keyboard[hmi_request::keyboard_layout].asInt())) {
+      return keyboard[hmi_response::num_configurable_keys].asUInt();
+    }
+  }
+
+  return 0;
+}
+
+bool SetGlobalPropertiesRequest::AreCustomizableKeysAllowed() const {
+  SDL_LOG_AUTO_TRACE();
+
+  const smart_objects::SmartObject& msg_params =
+      (*message_)[strings::msg_params];
+
+  if (!msg_params.keyExists(strings::keyboard_properties)) {
+    SDL_LOG_WARN("Keyboard properties are not available");
+    return true;
+  }
+
+  if (!msg_params[strings::keyboard_properties].keyExists(
+          hmi_request::customize_keys)) {
+    SDL_LOG_WARN("Customizable keys are not available");
+    return true;
+  }
+
+  auto custom_keys_array =
+      msg_params[strings::keyboard_properties][hmi_request::customize_keys]
+          .asArray();
+  if (custom_keys_array) {
+    uint32_t requested_key_count = custom_keys_array->size();
+    uint32_t allowed_key_count = GetAllowedNumberOfConfigurableKeys();
+
+    if (requested_key_count > allowed_key_count) {
+      return false;
+    }
```

Comment:
Should we be checking if the keyboard layout is supported as well?

Reply:
If requested keyboard layout is not supported, method `GetAllowedNumberOfConfigurableKeys()` returns `allowed_key_count` = 0. This happens because `GetAllowedNumberOfConfigurableKeys() `invokes `GetKeyboardLayout()` method, which returns requested keyboard layout, and also checks whether this layout is supported (compares it with an array of supported layouts).
So actually mentioned check is performed, but not directly (in nested method).